app/parsers/bulkrax/csv_parser.rb

```diff
@@ -8,14 +8,26 @@ def self.export_supported?
       true
     end
 
+    def records(_opts = {})
+      file_for_import = only_updates ? parser_fields['partial_import_file_path'] : import_file_path
+      # data for entry does not need source_identifier for csv, because csvs are read sequentially and mapped after raw data is read.
+      csv_data = entry_class.read_data(file_for_import)
+      importer.parser_fields['total'] = csv_data.count
+      importer.save
+      @records ||= csv_data.map { |record_data| entry_class.data_for_entry(record_data, nil) }
+    end
+
     def collections
-      # does the CSV contain a collection column?
-      return [] unless (import_fields & [:collection, :collections]).any?
+      col_mapping = Bulkrax.collection_field_mapping[self.entry_class.to_s]&.to_sym || :collection
+      model_mappings = Bulkrax.field_mappings[self.class.to_s]&.dig('model', :from) || []
```

Comment:
the `self.class.to_s` call in csv_entry.rb evaluations to "Class". will this not do the same?

Reply:
It shouldn't. This is an instance method, so `self` in this case is an instance of `CsvParser`: 
```ruby
self
=> #<Bulkrax::CsvParser:0x00005602f005e2b0 ...
self.class
=> Bulkrax::CsvParser
```

I think the method in the `CsvEntry` as a class method, so `self` in that case would be the class: 
```ruby
self
=> Bulkrax::CsvEntry
self.class
=> Class
```